File: DSA/Problems_Practice.cpp
// https://atcoder.jp/contests/agc005/tasks/agc005_a
#include<bits/stdc++.h>
#pragma GCC optimize ("O3")
#pragma GCC target ("sse4")
using namespace std;

#define FAST_IO ios_base::sync_with_stdio(false); cin.tie(NULL); cout.tie(NULL);

auto solve()
{
    string w; cin >> w;
    stack<char> arr;
    for (char x : w)
    {
        if (arr.size() == 0 or !(arr.top() == 'S' and x == 'T'))
        {
            arr.push(x);
        }
        else
        {
            arr.pop();
        }
    }
    cout << arr.size();
    return "\n";
}

int main()
{
    FAST_IO;
    cout << solve();
    return 0;
}

File: DSA/Problems_Practice.java
// https://atcoder.jp/contests/agc005/tasks/agc005_a
import java.io.*;
import java.lang.*;
import java.util.*;

public class Main { // atcoder - Main; codechef - Main;
	public static void main(String[] args) {
		Scanner scan = new Scanner(System.in);
		BufferedReader br = new BufferedReader(new InputStreamReader(System.in));
		char[] w = scan.nextLine().toCharArray();
		Stack<Character> stack = new Stack<>();
		for (Character x : w) {
			if ((stack.size() == 0) || !(Character.compare(stack.peek(), 'S') == 0 && Character.compare(x, 'T') == 0)) {
				stack.push(x);
			} else {
				stack.pop();
			}
		}
		int ans = stack.size();
		System.out.println(ans);
	}
}

